netket/experimental/driver/vmc_kernel.py

```diff
@@ -0,0 +1,164 @@
+    energy. Note that, to compute the updates, it is sufficient to invert an :math:`M\\times M` matrix
+    instead of a :math:`P\\times P` one. As a consequence, this formulation is useful
+    in the typical deep learning regime where :math:`P \gg M`.
+    
+    See R.Rende, L.L.Viteritti, L.Bardone, F.Becca and S.Goldt (https://arxiv.org/abs/2310.05715)
+    for a detailed description of the derivation. A similar result can be obtained by minimizing the
+    Fubini-Study distance with a specific constrain, see A.Chen and M.Heyl (https://arxiv.org/abs/2302.01941) 
+    for details.
+    """
+
+    def __init__(
+        self,
+        hamiltonian: AbstractOperator,
+        optimizer,
+        diag_shift,
+        *args,
+        linear_solver_fn: Callable[[jax.Array, jax.Array], jax.Array] = linear_solver,
+        jacobian_mode = None,
+        variational_state = None,
+        **kwargs,
+    ):
+        """
+        Initializes the driver class.
+
+        Args:
+            hamiltonian: The Hamiltonian of the system.
+            optimizer: Determines how optimization steps are performed given the
+                    bare energy gradient.
+            diag_shift: The diagonal shift of the stochastic reconfiguration matrix.
+                    Typical values are 1e-4 ÷ 1e-3.
+        """
+        super().__init__(hamiltonian, optimizer, variational_state=variational_state)
+
+        self._ham = hamiltonian.collect()  # type: AbstractOperator
+        _, self.unravel_params_fn = ravel_pytree(self.state.parameters)
+        self.diag_shift = diag_shift
+        self.jacobian_mode = jacobian_mode
+        self.linear_solver_fn = linear_solver_fn
+
+    def _forward_and_backward(self):
+        """
+        Performs a number of VMC optimization steps.
+
+        Args:
+            n_steps (int): Number of steps to perform.
+        """
+
+        self.state.reset()
+
+        # Compute the local energy estimator and average Energy
+        local_energy = self.state.local_estimators(self._ham).squeeze()
+        
+        e_mean = local_energy.mean()
+        self._loss_stats = e_mean
+
+        de = jnp.conj(local_energy - e_mean)
+
+        if self.jacobian_mode is None:
+            mode = "complex"
+        else:
+            #* mode='complex' is the most general; mode='holomorphic' could be implemented
+            assert self.jacobian_mode in ["complex", "real"], "Jacobian mode must be 'complex' or 'real'"
+            mode = self.jacobian_mode
+
+        jacobians = nkjax.jacobian(self.state._apply_fun,
+                                self.state.parameters,
+                                self.samples.squeeze(), 
+                                self.state.model_state,
+                                mode=mode,
+                                dense=True,
+                                center=True) #* jaxcobians is centered
+
+        updates = kernel_SR(jacobians, de, self.diag_shift, mode, self.linear_solver_fn)
```

Comment:
let's make this internal for now...
```suggestion
        updates = kernel_SR(jacobians, de, self.diag_shift, self.jacobian_mode, self._linear_solver_fn)
```